cli/src/commands/sign.rs

```diff
@@ -0,0 +1,71 @@
+use std::io::Write;
+
+use jj_lib::object_id::ObjectId;
+use jj_lib::signing::SignBehavior;
+
+use crate::cli_util::{user_error, CommandError, CommandHelper, RevisionArg};
+use crate::ui::Ui;
+
+/// Cryptographically sign a revision
+#[derive(clap::Args, Clone, Debug)]
+pub struct SignArgs {
+    /// What key to use, depends on the configured signing backend.
+    #[arg()]
+    key: Option<String>,
+    /// What revision to sign
+    #[arg(long, short, default_value = "@")]
+    revision: RevisionArg,
+    /// Sign the commit that is not authored by you or was already signed.
+    #[arg(long, short)]
+    force: bool,
+    /// Drop the signature, explicitly "un-signing" the commit.
+    #[arg(long, short = 'D', conflicts_with = "force")]
+    drop: bool,
+}
+
+pub fn cmd_sign(ui: &mut Ui, command: &CommandHelper, args: &SignArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+
+    let commit = workspace_command.resolve_single_rev(&args.revision, ui)?;
+    workspace_command.check_rewritable([&commit])?;
+
+    if !args.force {
+        if !args.drop && commit.is_signed() {
+            return Err(user_error(
+                "Commit is already signed, use --force to sign anyway",
+            ));
+        }
+        if commit.author().email != command.settings().user_email() {
+            return Err(user_error(
+                "Commit is not authored by you, use --force to sign anyway",
+            ));
+        }
+    }
+
+    let mut tx = workspace_command.start_transaction();
+
+    let behavior = if args.drop {
+        SignBehavior::Drop
+    } else if args.force {
+        SignBehavior::Force
+    } else {
+        SignBehavior::Own
+    };
+    let rewritten = tx
+        .mut_repo()
+        .rewrite_commit(command.settings(), &commit)
+        .override_sign_key(args.key.clone())
+        .set_sign_behavior(behavior)
+        .write()?;
+
+    tx.finish(ui, format!("sign commit {}", commit.id().hex()))?;
+
+    let summary = workspace_command.format_commit_summary(&rewritten);
+    if args.drop {
+        writeln!(ui.stderr(), "Signature was dropped: {summary}")?;
+    } else {
+        writeln!(ui.stderr(), "Commit was signed: {summary}")?;
+    }
+
+    Ok(())
+}
```

Comment:
It would be nice to have tests for this. Can we add a simple fake signing backend for tests? (I haven't thought much about it. Maybe it's harder than it seems?)

Reply:
Ok I have added a `MockSigningBackend` implementation. This gets used when `signing.backend = "mock"` is set in config. I don't love this as technically end users would be able to set this (though it is undocumented). I couldn't think of a better way to do this without reworking how the command tests are constructed but perhaps you know of a way - Let me know.

Assuming this is fine I have then added some `sign` command tests which use this mock backend.

Reply:
But there was a test signing backend somewhere I think?.